actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -208,32 +208,36 @@ def cache_unless(condition, name = {}, options = {}, &block)
       #
       # The digest will be generated using +virtual_path:+ if it is provided.
       #
-      def cache_fragment_name(name = {}, skip_digest: nil, virtual_path: nil)
+      def cache_fragment_name(name = {}, skip_digest: nil, virtual_path: nil, digest_path: nil)
         if skip_digest
           name
         else
-          fragment_name_with_digest(name, virtual_path)
+          virtual_path ||= @virtual_path
+          fragment_name_with_digest(name, virtual_path, digest_path)
         end
       end
 
     private
 
-      def fragment_name_with_digest(name, virtual_path)
-        virtual_path ||= @virtual_path
+      def digest_path_from_virtual(virtual_path)
+        digest = Digestor.digest(name: virtual_path, finder: lookup_context, dependencies: view_cache_dependencies)
 
-        if virtual_path
-          name = controller.url_for(name).split("://").last if name.is_a?(Hash)
-
-          if digest = Digestor.digest(name: virtual_path, finder: lookup_context, dependencies: view_cache_dependencies).presence
-            [ "#{virtual_path}:#{digest}", name ]
-          else
-            [ virtual_path, name ]
-          end
+        if digest.present?
+          "#{virtual_path}:#{digest}"
         else
-          name
+          virtual_path
         end
       end
 
+      def fragment_name_with_digest(name, virtual_path, digest_path)
+        return name if !virtual_path && !digest_path
```

Comment:
> Do we have cases of nil virtual path but present digest path?

We don't but since i'm modifying the method signature of a public method `cache_fragment_name` other people can in theory end up here with a nil virtual path but a present digest_path.

> I'd tend to a conditional expression over a guard clause here too.

Sounds good.